### PR TITLE
Remove pr image on gh action test step

### PR DESCRIPTION
**Pull request by pacevedom**
Remove built images from quay after k8s e2e test is finished.

### CODE REVIEW

**Comment by pacevedom** [comment]
cc @ybettan

File: .github/workflows/k8s-e2e.yaml [hunk 1]
```diff
@@ -82,3 +82,21 @@ jobs:
     - run: kubectl apply -f  charts/example/centos-simple-kmod-0.0.1/centos-simple-kmod.yaml
     - run: sleep 360
     - run: kubectl logs `kubectl get pod -n openshift-special-resource-operator | grep special | awk '{ print $1 }'` -c manager -n openshift-special-resource-operator
+  remove:
+    name: remove-image
+    needs:
+      - build
```
Comment (ybettan) [inline .github/workflows/k8s-e2e.yaml]: Technically it only needs `e2e` right?! I think we can remove the `build`.

File: .github/workflows/k8s-e2e.yaml [hunk 2]
```diff
@@ -82,3 +82,21 @@ jobs:
...
+    concurrency:
+      group: ${{ github.workflow }}_e2e-test_$${ github.head_ref }}
+      cancel-in-progress: true
+    steps:
+    - uses: benjlevesque/short-sha@v1.2
+      id: short-sha
+      with:
+        length: 10
+    - run: |
+        curl -s -X GET -H "Authorization: Bearer ${{ secrets.QUAY_API_TOKEN }}" https://quay.io/api/v1/repository/openshift-psap/special-resource-operator/tag/?specificTag=pr-${{ steps.short-sha.outputs.sha }}
```
Comment (ybettan) [inline .github/workflows/k8s-e2e.yaml]: Why do we need the get?

File: .github/workflows/k8s-e2e.yaml [hunk 3]
```diff
@@ -82,3 +82,21 @@ jobs:
...
+    - run: |
+        curl -s -X DELETE -H "Authorization: Bearer ${{ secrets.QUAY_API_TOKEN }}" https://quay.io/api/v1/repository/openshift-psap/special-resource-operator/tag/pr-${{ steps.short-sha.outputs.sha }}
```
Comment (ybettan) [inline .github/workflows/k8s-e2e.yaml:98]: Did you check if it is working? I am not sure evaluating GH values like `steps.short-sha.outputs.sha` is working inside `run` commands, I think you may need to add `env:\n\tTAG=${{ steps.short-sha.outputs.sha }}` and then use `TAG` inside the the `run` step

File: .github/workflows/k8s-e2e.yaml (hunk 1 above)
Comment (pacevedom) [reply .github/workflows/k8s-e2e.yaml]: You are right. Updated.

File: .github/workflows/k8s-e2e.yaml (hunk 2 above)
Comment (pacevedom) [reply .github/workflows/k8s-e2e.yaml]: Good point, its not needed.

File: .github/workflows/k8s-e2e.yaml (hunk 3 above)
Comment (pacevedom) [reply .github/workflows/k8s-e2e.yaml:98]: It does work, substitutions work without an issue. Besides that, I had to create a fake organization and application for a token, but it does work:
`$ curl -s -X DELETE -H "Authorization: Bearer ***" https://quay.io/api/v1/repository/pacevedo/special-resource-operator/tag/master -i
HTTP/2 204 
date: Mon, 10 Jan 2022 07:54:46 GMT
server: nginx/1.12.1
access-control-allow-origin: *
access-control-allow-methods: PUT, OPTIONS, DELETE
access-control-max-age: 21600
access-control-allow-headers: AUTHORIZATION, CONTENT-TYPE, X-REQUESTED-WITH
x-frame-options: DENY
strict-transport-security: max-age=63072000; preload`

Comment (ybettan) [reply .github/workflows/k8s-e2e.yaml:98]: OK

**Comment by ybettan** [comment]
WDYT about putting that flow into a separate workflow that is triggered closing the PR? Something like:
```name: pr-clean

on:
  pull_request:
    types: [closed]
```

I am just wondering what is the best approach here.

By deleting only once, we leverage the layer caching of quay.io, other wise we will push/pull the entire image on every run.

**Comment by ybettan** [comment]
I think we don't have to push images at all, using k3d we can build the images locally and inject them into k3d.
Testing it.

**Comment by zvonkok** [comment]
You can do this with kind as well: https://kind.sigs.k8s.io/docs/user/quick-start/#loading-an-image-into-your-cluster
I would propose to stick with upstream vanilla kube bits and not use external kube variants.

**Comment by ybettan** [comment]
@pacevedom Can we close that PR?
The only e2e job that we have after https://github.com/openshift-psap/special-resource-operator/pull/272 isn't pushing anyway to quay.io, so there is nothing to clean.

**Pull request closed without merging**